workspaces/orchestrator/plugins/orchestrator/src/components/WorkflowRunsTabContentFiltered.tsx

```diff
@@ -0,0 +1,165 @@
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+import React, { useState } from 'react';
+
+import {
+  ErrorPanel,
+  InfoCard,
+  Link,
+  SelectItem,
+  TableColumn,
+} from '@backstage/core-components';
+import {
+  useApi,
+  useRouteRef,
+  useRouteRefParams,
+} from '@backstage/core-plugin-api';
+
+import { Grid } from '@material-ui/core';
+
+import {
+  capitalize,
+  ellipsis,
+  ProcessInstanceState,
+  ProcessInstanceStatusDTO,
+} from '@red-hat-developer-hub/backstage-plugin-orchestrator-common';
+
+import { orchestratorApiRef } from '../api';
+import { DEFAULT_TABLE_PAGE_SIZE, VALUE_UNAVAILABLE } from '../constants';
+import usePolling from '../hooks/usePolling';
+import { workflowInstanceRouteRef, workflowRouteRef } from '../routes';
+import { Selector } from './Selector';
+import OverrideBackstageTable from './ui/OverrideBackstageTable';
+import { mapProcessInstanceToDetails } from './WorkflowInstancePageContent';
+import { WorkflowInstanceStatusIndicator } from './WorkflowInstanceStatusIndicator';
+import { WorkflowRunDetail } from './WorkflowRunDetail';
+
+const makeSelectItemsFromProcessInstanceValues = () =>
+  [
+    ProcessInstanceState.Active,
+    ProcessInstanceState.Error,
+    ProcessInstanceState.Completed,
+    ProcessInstanceState.Aborted,
+    ProcessInstanceState.Suspended,
+  ].map(
+    (status): SelectItem => ({
+      label: capitalize(status),
+      value: status,
+    }),
+  );
+
+export const WorkflowRunsTabContentFiltered = () => {
+  const { workflowId } = useRouteRefParams(workflowRouteRef);
+
+  const orchestratorApi = useApi(orchestratorApiRef);
+  const workflowInstanceLink = useRouteRef(workflowInstanceRouteRef);
+  const [statusSelectorValue, setStatusSelectorValue] = useState<string>(
+    Selector.AllItems,
+  );
+
+  const fetchInstances = React.useCallback(async () => {
+    // TODO: use pagination with generic permission, skip (or use FE-only) for specific permissions
+    const instances = await orchestratorApi.listInstances({});
+    const clonedData: WorkflowRunDetail[] =
+      instances.data.items?.map(mapProcessInstanceToDetails) || [];
+    return clonedData.filter(item => item.workflowId === workflowId);
+  }, [orchestratorApi, workflowId]);
+
+  const { loading, error, value } = usePolling(fetchInstances);
+
+  const columns = React.useMemo(
+    (): TableColumn<WorkflowRunDetail>[] => [
+      {
+        title: 'ID',
+        field: 'id',
+        render: data => (
+          <Link to={workflowInstanceLink({ instanceId: data.id })}>
+            {ellipsis(data.id)}
```

Comment:
The UUID length is constant, we do not need to shorten it in the majority of cases.
For very narrow displays, we can play with the CSS `overflow`.

---
Recently we are wasting the space:

![Screenshot From 2024-12-13 09-06-35](https://github.com/user-attachments/assets/d8d36f9f-2a24-4246-a8ee-ad6377f77208)

Reply:
@mareklibra 
the instance ids are rather long and there's no need to see the whole thing
This is also unrelated to this code change
![image](https://github.com/user-attachments/assets/c768bafd-ad32-4be8-b75e-afc2236a26a7)

Reply:
> the instance ids are rather long and there's no need to see the whole thing

In that case we do not need to show it at all.

Reply:
what should we show instead? What should the user do if he wants to search for a specific instance?

Reply:
This is out of the scope of this PR, needs to be discussed with UX

Reply:
I have noticed this issue since the code is being touched here. But agree, we can move it to another issue.

IMO, we should show the user either correct and full info or nothing. What is a fragment of the UUID for?

If the user is not interested in the UUID (which is the case IMO), we don't need to show it. And link the instance detail through the Workflow name or Started column. The UUID can be shown on the details page.

I think we can move the Started column to the first position and make a link out of it.

I will open a Jira issue.